simulation/agent.py

```diff
@@ -271,9 +324,22 @@ def on_vision_update(self) -> None:
         pass
 
     @abstractmethod
-    def on_noise(self, noise: 'world.NoiseEvent') -> None:
-        """ Noise handler, will be called before `on_tick` """
-        pass
+    def on_noise(self, noises: List['world.NoiseEvent']) -> None:
```

Comment:
This method should have no implementation, its only purpose is to be a "placeholder" for the code in the actual agents (like java interfaces). This should probably all go in the `Agent.tick` method.